lib/node_modules/@stdlib/math/base/special/rempio2/src/main.c

```diff
@@ -0,0 +1,535 @@
+static const int32_t THREE_PIO4_HIGH_WORD = 0x4002d97c;
+
+// High word for 5π/4: 0x400f6a7a = 1074752122 => 01000000000011110110101001111010
+static const int32_t FIVE_PIO4_HIGH_WORD = 0x400f6a7a;
+
+// High word for 6π/4: 0x4012d97c = 1074977148 => 01000000000100101101100101111100
+static const int32_t THREE_PIO2_HIGH_WORD = 0x4012d97c;
+
+// High word for 7π/4: 0x4015fdbc = 1075183036 => 01000000000101011111110110111100
+static const int32_t SEVEN_PIO4_HIGH_WORD = 0x4015fdbc;
+
+// High word for 8π/4: 0x401921fb = 1075388923 => 01000000000110010010000111111011
+static const int32_t TWO_PI_HIGH_WORD = 0x401921fb;
+
+// High word for 9π/4: 0x401c463b = 1075594811 => 01000000000111000100011000111011
+static const int32_t NINE_PIO4_HIGH_WORD = 0x401c463b;
+
+// 2^20*π/2 = 1647099.3291652855 => 0100000100111001001000011111101101010100010001000010110100011000 => high word => 0x413921fb = 1094263291 => 01000001001110010010000111111011
+static const int32_t MEDIUM = 0x413921fb;
+
+// Exponent mask (2047 => 0x7ff):
+static const int32_t EXPONENT_MASK = 0x7ff;
+
+
+/*
+* Table of constants for `2/π` (`396` hex digits, `476` decimal).
+*
+* Integer array which contains the (`24*i`)-th to (`24*i+23`)-th bit of `2/π` after binary point. The corresponding floating value is
+*
+* ```tex
+* \operatorname{ipio2}[i] \cdot 2^{-24(i+1)}
+* ```
+*
+* This table must have at least `(e0-3)/24 + jk` terms. For quad precision (`e0 <= 16360`, `jk = 6`), this is `686`.
+*/
+static const int32_t IPIO2[] = {
+	0xA2F983, 0x6E4E44, 0x1529FC, 0x2757D1, 0xF534DD, 0xC0DB62,
+	0x95993C, 0x439041, 0xFE5163, 0xABDEBB, 0xC561B7, 0x246E3A,
+	0x424DD2, 0xE00649, 0x2EEA09, 0xD1921C, 0xFE1DEB, 0x1CB129,
+	0xA73EE8, 0x8235F5, 0x2EBB44, 0x84E99C, 0x7026B4, 0x5F7E41,
+	0x3991D6, 0x398353, 0x39F49C, 0x845F8B, 0xBDF928, 0x3B1FF8,
+	0x97FFDE, 0x05980F, 0xEF2F11, 0x8B5A0A, 0x6D1F6D, 0x367ECF,
+	0x27CB09, 0xB74F46, 0x3F669E, 0x5FEA2D, 0x7527BA, 0xC7EBE5,
+	0xF17B3D, 0x0739F7, 0x8A5292, 0xEA6BFB, 0x5FB11F, 0x8D5D08,
+	0x560330, 0x46FC7B, 0x6BABF0, 0xCFBC20, 0x9AF436, 0x1DA9E3,
+	0x91615E, 0xE61B08, 0x659985, 0x5F14A0, 0x68408D, 0xFFD880,
+	0x4D7327, 0x310606, 0x1556CA, 0x73A8C9, 0x60E27B, 0xC08C6B
+};
+
+// Double precision array, obtained by cutting `π/2` into `24` bits chunks...
+static const double PIO2[] = {
+	1.57079625129699707031e+00, // 0x3FF921FB, 0x40000000
+	7.54978941586159635335e-08, // 0x3E74442D, 0x00000000
+	5.39030252995776476554e-15, // 0x3CF84698, 0x80000000
+	3.28200341580791294123e-22, // 0x3B78CC51, 0x60000000
+	1.27065575308067607349e-29, // 0x39F01B83, 0x80000000
+	1.22933308981111328932e-36, // 0x387A2520, 0x40000000
+	2.73370053816464559624e-44, // 0x36E38222, 0x80000000
+	2.16741683877804819444e-51  // 0x3569F31D, 0x00000000
+};
+
+// Arrays for storing temporary values:
+double TX[ 3 ] = { 0.0 };
+double TY[ 2 ] = { 0.0 };
+double F[ 20 ];
+double Q[ 20 ];
+double FQ[ 20 ];
+int32_t IQ[ 20 ];
```

Comment:
These need to be moved to within the implementation. In C, we need to worry about thread safety. See, e.g., https://svnweb.freebsd.org/base/release/12.2.0/lib/msun/src/k_rem_pio2.c?view=markup#l297.